packages/block-library/src/heading/block.json

```diff
@@ -30,49 +30,15 @@
 		},
 		"fontSize": true,
 		"lineHeight": true,
-		"__experimentalSelector": {
-			"core/heading/h1": {
-				"selector": "h1",
-				"title": "h1",
-				"attributes": {
-					"level": 1
-				}
-			},
-			"core/heading/h2": {
-				"selector": "h2",
-				"title": "h2",
-				"attributes": {
-					"level": 2
-				}
-			},
-			"core/heading/h3": {
-				"selector": "h3",
-				"title": "h3",
-				"attributes": {
-					"level": 3
-				}
-			},
-			"core/heading/h4": {
-				"selector": "h4",
-				"title": "h4",
-				"attributes": {
-					"level": 4
-				}
-			},
-			"core/heading/h5": {
-				"selector": "h5",
-				"title": "h5",
-				"attributes": {
-					"level": 5
-				}
-			},
-			"core/heading/h6": {
-				"selector": "h6",
-				"title": "h6",
-				"attributes": {
-					"level": 6
-				}
-			}
+		"__experimentalSelector": "h1,h2,h3,h4,h5,h6",
+		"__experimentalSelectorsForElements": {
+			"link": "h1 a,h2 a,h3 a,h4 a,h5 a,h6 a",
+			"h1": "h1",
+			"h2": "h2",
+			"h3": "h3",
```

Comment:
When I have a heading block at run time in the editor, how do I know if the heading is being affected by the h1 selector or h2 selector?
Previously we were using an attributes specification so at run time we could map a block instance to a selector:
```
				"attributes": {
					"level": 6
				}
```

Reply:
I've updated the PR description to share more details about this, also in the issue this fixes there's more context about the rationale.

The quick answer is: no, we no longer need to do that because the heading block maps to a single UI panel in the global styles sidebar (as opposed to the current situation in which it has one panel _per sub-selector_). Elements are meant to absorb this complexity but are not yet formalized at the UI level ([see](https://github.com/WordPress/gutenberg/issues/27473#issuecomment-796773866))

Reply:
> The quick answer is: no, we no longer need to do that because the heading block maps to a single UI panel in the global styles sidebar (as opposed to the current situation in which it has one panel per sub-selector). Elements are meant to absorb this complexity but are not yet formalized at the UI level (see)


If a theme sets different font sizes per heading level. I'm on the block editor, outside of global styles and I have a heading block, how does the editor know what is the default font size of that heading block to show on the block level inspector?

Reply:
The data is still structured in a way the editor could use it if/when we do that (in other words, that the data lives at `styles.elements.h2.color.etc` instead of `core/heading/h2.styles.color.etc` shouldn't be a blocker).

Reply:
Yes, we have the data but it seems now given a specific block instance e.g: a heading 2 block the editor will not be able to know that styles should be loaded from styles.elements.h2.color.etc. There is nothing relating that a level 2 heading attribute should make styles load from styles.elements.h2 and a level 3 attribute should make styles load styles.elements.h3. So given a block instance, the block inspector on the post editor will not know the defaults.

Reply:
To me, this is one of those things that we should revisit when we cross that bridge: when we have more info about elements in the UI, how well this works "by default" without [allowing blocks to re-define elements](https://github.com/WordPress/gutenberg/pull/30541#discussion_r621150182), etc. I fully trust our ability to implement that matching _if_ we need it at that point.